### PR TITLE
Update CppCoreGuidelines.md

### PR DESCRIPTION
**Pull request by ritikpal1122**

### CODE REVIEW

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -511,7 +511,7 @@ This loop is a restricted form of `std::find`:
         cin >> val;
         // ...
         int index = -1;                    // bad, plus should use gsl::index
-        for (int i = 0; i < v.size(); ++i) {
+        for (int i = 0; i < v.size(); i++) {
```
Comment (beinhaerter) [inline CppCoreGuidelines.md:514]: Why? What is the rationale behind this change?
Compare https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p9-dont-waste-time-or-space and its Enforcement section.

**Comment by jwakely** [comment]
This is not an improvement.
